adr/0003-multiple-templates-per-component.md

```diff
@@ -0,0 +1,52 @@
+# 1. Allow multiple templates 
+
+Date: 2021-08-30
+
+## Status
+
+Proposed.
+
+## Context
+
+As components become larger (for example, because you are implementing a whole page), it becomes
+useful to be able to extract sections of the view to a different file. ActionView has
+partials, and ViewComponent lacks a similar mechanism. 
+
+ActionView partials have the problem that their interface is not introspectable. Data
+may be passed into the partial via ivars or locals, and it is impossible to know
+which without actually opening up the file. Additionally, partials are globally 
+invocable, thus making it difficult to detect if a given partial is in use or not,
+and who are its users.
+
+An option would be to extract another Component for the extracted section, but that
+has the following drawbacks:
+
+1. It creates a new public class.
+2. If new component is invoked repeatedly (eg, for a list of items), this creates
+   GC pressure by creating lots of intermediate objects.
+
```

Comment:
```suggestion
## Considered Options

* Introduce component partials to components
* Keep components as-is

### Component partials

Quick description...

**Pros:**
* Better performance due to lack of GC pressure and object creation

**Cons:**
* Another concept for users of ViewComponent to learn and understand.
* Components are no longer the only way to encapsulate behavior.
* Partials aren't reusable, and belong to a specific component.

### Keeping components as-is

**Pros:**
* The API remains simple and components are the only way to encapsulate behavior.

**Cons:**
* Extracting a component results in more GC and intermediate objects.
```

I've used this format in the past to help with ADR's, thoughts on adding this to help us get a quick understanding of each option? I haven't completely filled out each pro/con section fwiw, but I think that leaves room to suggest changes and discuss individual points. (e.g. your garbage collection/object concern is a very valid, and very good point)

Reply:
I like the format! I have omitted the "Partials aren't reusable, and belong to a specific component.", since I actually think that is a Pro, not a Con. Perhaps we can start a new thread to discuss that?